src/app/services/featureLimit.js

```diff
@@ -7,8 +15,65 @@ module.exports = (Model, App) => {
     });
   };
 
+  const getIndividualFreeTier = async () => {
+    return getTierByPlanId(INDIVIDUAL_FREE_TIER_PLAN_ID);
+  };
+
+  const findLimitByLabelAndTier = async (tierId, label) => {
+    return Model.limits.findOne({
+      where: {
+        label,
+      },
+      include: [
+        {
+          model: Model.tiers,
+          where: {
+            id: tierId,
+          },
+        },
+      ],
+    });
+  };
+
+  const isBooleanLimitNotAvailable = (limit) => {
+    return limit.type === LimitTypes.Boolean && limit.value !== 'true';
+  };
+
+  const shouldLimitBeEnforced = async (user, limitLabel, data) => {
+    const limit = await findLimitByLabelAndTier(user.tierId, limitLabel);
+
+    if (!limit) {
+      throw new NoLimitFoundForUserTierAndLabel('No limit found for this user tier and label!');
+    }
+    if (limit.type === LimitTypes.Boolean) {
+      return isBooleanLimitNotAvailable(limit);
+    }
+
+    const isLimitSuprassed = await checkCounterLimit(user, limit, data);
+
+    return isLimitSuprassed;
+  };
+
+  const checkCounterLimit = (user, limit, data) => {
+    switch (limit.label) {
+      case LimitLabels.MaxFileUploadSize:
+        return isMaxFileSizeLimitSurprassed({ limit, data });
+      default:
+        return false;
+    }
+  };
+
+  const isMaxFileSizeLimitSurprassed = async ({ limit, data }) => {
+    const {
+      file: { size },
+    } = data;
+    return Number(limit.value) < size / 1024 / 1024 / 1024;
```

Comment:
We are setting the limit value in bytes, why is the division between 2^30 needed? The size is received in bytes also